src/TriggerService/TriggerHostedService.cs

```diff
@@ -327,11 +331,27 @@ public async Task<ProcessedWorkflowItem> GetBlobFileNameAndData(string url)
 
             if (GetBlockBlobStorage(url, storageAccounts) is IAzureStorage aStorage)
             {
-                data = await aStorage.DownloadBlockBlobAsync(url);
+                try
+                {
+                    data = await aStorage.DownloadBlockBlobAsync(url);
+                }
+                catch (Exception e)
+                {
+                    throw new Exception($"{e.InnerException} while processing blob, cannot find {url} in {aStorage.AccountName}. " +
```

Comment:
e?.InnerException in case it's null

Reply:
How do we know the exception is "notfound" here?  Seems like it could be one of many issues

Reply:
Two things to consider: replace `e.InnerException` with `e?.InnerException.Message ?? e.Message` (in both places) and instead of knowing what the failure was, simply state "failed to retrieve {url} from {accountname}"

Reply:
[I couldn't find specific exceptions](https://learn.microsoft.com/en-us/dotnet/api/microsoft.azure.storage.blob.cloudblob.downloadtostreamasync?view=azure-dotnet) kicked off by `CloudBlob.DownloadToStreamAsync` (since that's the only thing that could throw errors in `DownloadBlockBlobAsync`) so I tried to thread the needle on a message that'd be a little more verbose but still include the exception. I like your suggestion @BMurri if you think that'd be enough information to solve the original bug

Reply:
Two things: 1) please add `e.GetType().FullName` in an appropriate place in the message, and 2) add `e` to the constructor of `Exception`, so that the captured and reported exception is the `InnerException` of the exception that we will ultimately log later on. That will help later in reviewing logs when this happens.

Applies for both instances.

Reply:
Updated and working. Thank you so much for the feedback!